src/query/interpret/eval.hpp

```diff
@@ -847,6 +850,9 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
       }
       auto res = function.function_(arguments, function.arguments_.size(), function_ctx);
       MG_ASSERT(res.GetMemoryResource() == ctx_->memory);
+      if (!is_transactional && res.ContainsDeleted()) [[unlikely]] {
+        return TypedValue(ctx_->memory);
+      }
```

Comment:
Can we refactor bit this Visit function, I think only arguments allocation and processing (for loop) needs to be in separate if-else blocks, otherwise we can isolate function call, mg_assert and if check for deleted to be of common execution  part

Reply:
Done (with the caveat that the function call remains inside the if-else blocks)